Update documentation

neon60 opened the pull request.

neon60 commented:
I have to test the rocm-smi build and install, because my changes effecting that too.

neon60 commented:
@saadrahim please check this PR and give some feedbacks. I think we have to split up the readme file.

dmitrii-galantsev commented:
@saadrahim @samjwu Please have a look. The docs changes are a little over my head :)

MathiasMagnus reviewed (approved):
The resulting docs look fine.

I think the difference in doc style may warrant a spinoff issue to unify the style of the Python API and the C++ API (moving the C++ API over to Breathe as opposed to the current doxysphinx).

samjwu commented on docs/python_api.rst:
```diff
@@ -0,0 +1,265 @@
+Python API Reference
+====================
+
+This chapter describes the rocm_smi Python module API.
```
```suggestion
This chapter describes the ROCm SMI Python module API.
```

samjwu commented on docs/c++_tutorials.rst:
```diff
@@ -0,0 +1,32 @@
+C++ Tutorials
+====================
+
+This chapter is the rocm_smi C++ api tutorials.
```
```suggestion
This chapter contains the ROCm SMI C++ API tutorials.
```

samjwu reviewed (approved):
Looks good to me.

Had a couple suggestions: I see the README spells the project name as ROCm SMI

neon60 commented:
@dmitrii-galantsev I had to split up the python_smi_tools/rsmiBindings.py file to two separate files:
- python_smi_tools/rsmiBindings.py
- python_smi_tools/rsmiBindingsInit.py

Is it okay with you and your team? Can you check the generated documentation? Do you have any feedbacks?
We would like to merge this in, one day :)

charis-poag-amd commented on docs/sphinx/requirements.txt:
```diff
@@ -116,7 +116,7 @@ requests==2.28.2
     #   pygithub
     #   sphinx
-rocm-docs-core[api_reference]>=0.30.3
+rocm-docs-core[api_reference]>=0.31.0
```
~~Should this also be 30.3? Or is there a reason for bumping up to 31.0?~~
Disregard - large enough change to bump up for. 👍

neon60 replied:
@charis-poag-amd Can you merge this PR in here? I would like to avoid uploading via Gerrit?

neon60 committed:
Update docuemnation and add python API documentation

Add tutorials
Update python code comment
Split up the rsmiBindinsg.py
Rebased the work

neon60 committed:
Add one line

neon60 committed:
Fix warnings

neon60 committed:
Add Changelog

dmitrii-galantsev reviewed (approved):
Thank you for your work! Sorry it took a while to respond. I will approve here and merge internally. Should propagate in the next day or 2!

dmitrii-galantsev commented:
Merged: https://github.com/ROCm/rocm_smi_lib/commit/50a079af0f059ddfafe27e908f4878d759891eca
